### PR TITLE
Prevent pause menu if video isn't in a ready state

### LINKED ISSUE #1505: Pausing Before Playback Freezes Client
## Software Versions
- Jellyfin Server Version:  10.8.12
- Roku Client Version:  [Jellyfin-Roku-dev-3a80cdea3cdee3d540b7dde71ed22d6ac81f2174](https://github.com/jellyfin/jellyfin-roku/suites/18196517254/artifacts/1050385346)

## Describe the bug
Pausing playback before playback has actually begun completely freezes the client. The spinner stops spinning, the Pause Menu cannot be navigated, the clock does not update, and no buttons on the remote do anything (aside from Home and Power).

## How To Reproduce
1. Press Play on a Show or Movie
2. Press Pause after the black screen and spinner are showing but before media playback has begun
3. Bug occurs

## Expected behavior
Video loads, paused at the first frame.
Pause Menu is navigable.
Remote buttons work (play, back, volume, etc.).

## Logs
Nothing useful seen in server log.
Not sure how to access any client logs.

## Screenshots
After pressing pause, the screen is stuck at:
image
Media Info at the top intentionally grayed out

## Connection Information
Local server with HTTP only behind a reverse proxy handling HTTPS

## Additional context
Seen with direct play Shows and Movies
Not seen with transcoded Live TV or Recordings

### PR DESCRIPTION
**Pull request by 1hitsong**
## Changes
Checks if video state is in an acceptable state to show the pause menu prior to showing it.

## Issues
Fixes #1505

### CODE REVIEW

File: components/video/VideoPlayerView.bs [hunk 1]
```diff
@@ -568,6 +568,14 @@ sub bufferCheck(msg)
 
 end sub
 
+' stateSupportsPauseMenu: Check if current video state allows showing the pause menu
```
Comment (cewert) [inline components/video/VideoPlayerView.bs]:
```suggestion
' stateAllowsPauseMenu: Check if current video state allows showing the pause menu
```

File: components/video/VideoPlayerView.bs [hunk 2]
```diff
@@ -620,6 +628,9 @@ function onKeyEvent(key as string, press as boolean) as boolean
 
     if key = "down" and not m.top.trickPlayBar.visible
         if not m.LoadMetaDataTask.isIntro
+            ' Don't allow user to open menu prior to video loading
+            if not stateAllowsPauseMenu() then return true
```
Comment (cewert) [inline components/video/VideoPlayerView.bs:632]:
```suggestion
            if not stateAllowsPauseMenu() then return false
```
we should only return true if we handled the key press

File: components/video/VideoPlayerView.bs [hunk 3]
```diff
@@ -628,6 +639,9 @@ function onKeyEvent(key as string, press as boolean) as boolean
 
     else if key = "up" and not m.top.trickPlayBar.visible
         if not m.LoadMetaDataTask.isIntro
+            ' Don't allow user to open menu prior to video loading
+            if not stateAllowsPauseMenu() then return true
```
Comment (cewert) [inline components/video/VideoPlayerView.bs:643]:
```suggestion
            if not stateAllowsPauseMenu() then return false
```

File: components/video/VideoPlayerView.bs [hunk 4]
```diff
@@ -636,6 +650,9 @@ function onKeyEvent(key as string, press as boolean) as boolean
 
     else if key = "OK" and not m.top.trickPlayBar.visible
         if not m.LoadMetaDataTask.isIntro
+            ' Don't allow user to open menu prior to video loading
+            if not stateAllowsPauseMenu() then return true
```
Comment (cewert) [inline components/video/VideoPlayerView.bs:654]:
```suggestion
            if not stateAllowsPauseMenu() then return false
```

File: components/video/VideoPlayerView.bs [hunk 5]
```diff
@@ -649,6 +666,10 @@ function onKeyEvent(key as string, press as boolean) as boolean
     ' Disable pause menu for intro videos
     if not m.LoadMetaDataTask.isIntro
         if key = "play" and not m.top.trickPlayBar.visible
+
+            ' Don't allow user to open menu prior to video loading
+            if not stateAllowsPauseMenu() then return true
```
Comment (cewert) [inline components/video/VideoPlayerView.bs:671]:
```suggestion
            if not stateAllowsPauseMenu() then return false
```

**Review by cewert — CHANGES_REQUESTED** [review]

File: components/video/VideoPlayerView.bs (hunk 3 above)
Comment (1hitsong) [reply components/video/VideoPlayerView.bs:643]: Passing true was by design. We don't want to allow the keypress to pass through, it causes the trickplay bar to display in confusing state.

Comment (cewert) [reply components/video/VideoPlayerView.bs:643]: Good call. I didn't realize the trickplay bar would pick up the keypress

**Review by cewert — APPROVED** [review]

**Pull request merged**